cmd/probe/tcp_probe_test.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+)
+
+func TestTCPProbe(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), false)
+
+		if err := p.Run(t.Context()); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	t.Run("conn should fail", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
+
+		if err := p.Run(t.Context()); !errors.Is(err, errConnectionSucceeded) {
+			t.Fatalf("expecting error %v, got %v", errConnectionSucceeded, err)
+		}
+	})
+
+	t.Run("conn should not fail", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
+
+		l.Close()
+
+		if err := p.Run(t.Context()); !errors.Is(err, errConnectionFailed) {
+			t.Fatalf("expecting error %v, got %v", errConnectionFailed, err)
+		}
+	})
+
+	t.Run("context aware", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
+
+		ctx, cancel := context.WithCancel(t.Context())
+		cancel()
+
+		if err := p.Run(ctx); !errors.Is(err, context.Canceled) {
+			t.Fatalf("expecting error %v, got %v", context.Canceled, err)
```

Comment:
This test is not generating the expected error for me, I get this:
```
--- FAIL: TestTCPProbe (0.00s)
    --- FAIL: TestTCPProbe/context_aware (0.00s)
        tcp_probe_test.go:64: expecting error context canceled, got <nil>
FAIL
```

Reply:
Guess who forgot to add some changes before pushing? 🤦🏽